segmentstore/server/src/main/java/io/pravega/segmentstore/server/tables/ContainerTableExtensionImpl.java

```diff
@@ -0,0 +1,208 @@
+    }
+
+    @Override
+    public CompletableFuture<Void> merge(@NonNull String targetSegmentName, @NonNull String sourceSegmentName, Duration timeout) {
+        Exceptions.checkNotClosed(this.closed.get(), this);
+        throw new UnsupportedOperationException();
+    }
+
+    @Override
+    public CompletableFuture<Void> seal(String segmentName, Duration timeout) {
+        Exceptions.checkNotClosed(this.closed.get(), this);
+        throw new UnsupportedOperationException();
+    }
+
+    @Override
+    public CompletableFuture<List<Long>> put(@NonNull String segmentName, @NonNull List<TableEntry> entries, Duration timeout) {
+        Exceptions.checkNotClosed(this.closed.get(), this);
+        throw new UnsupportedOperationException();
+    }
+
+    @Override
+    public CompletableFuture<Void> remove(@NonNull String segmentName, @NonNull Collection<TableKey> keys, Duration timeout) {
+        Exceptions.checkNotClosed(this.closed.get(), this);
+        throw new UnsupportedOperationException();
+    }
+
+    @Override
+    public CompletableFuture<List<TableEntry>> get(@NonNull String segmentName, @NonNull List<ArrayView> keys, Duration timeout) {
+        Exceptions.checkNotClosed(this.closed.get(), this);
+        throw new UnsupportedOperationException();
+    }
+
+    @Override
+    public CompletableFuture<AsyncIterator<IteratorItem<TableKey>>> keyIterator(@NonNull String segmentName, IteratorState continuationToken,
+                                                                                Duration timeout) {
+        Exceptions.checkNotClosed(this.closed.get(), this);
+        throw new UnsupportedOperationException();
+    }
+
+    @Override
+    public CompletableFuture<AsyncIterator<IteratorItem<TableEntry>>> entryIterator(@NonNull String segmentName, IteratorState continuationToken,
+                                                                                    Duration timeout) {
+        Exceptions.checkNotClosed(this.closed.get(), this);
+        throw new UnsupportedOperationException();
+    }
+
+    //endregion
+
+    //region TableWriterConnector
+
+    @RequiredArgsConstructor
+    private class TableWriterConnectorImpl implements TableWriterConnector {
+        @Getter
+        private final SegmentMetadata metadata;
+
+        @Override
+        public EntrySerializer getSerializer() {
+            return ContainerTableExtensionImpl.this.serializer;
+        }
+
+        @Override
+        public KeyHasher getKeyHasher() {
+            return ContainerTableExtensionImpl.this.hasher;
+        }
+
+        @Override
+        public CompletableFuture<DirectSegmentAccess> getSegment(Duration timeout) {
+            return ContainerTableExtensionImpl.this.segmentContainer.forSegment(this.metadata.getName(), timeout);
+        }
+
+        @Override
+        public void notifyIndexOffsetChanged(long lastIndexedOffset) {
+            // Not yet implemented. Will be done in issue 2878.
```

Comment:
These should throw in the mean time, or at least log something loud so we know and don't forget

Reply:
In its current form, this class doesn't do anything. It's just a stub. The PR for #2878 (the code is out there, but no PR yet), will fill this class in with useful code. 

Theoretically if I have this throw an exception now, the `WriterTableProcessor` would not work, as it invokes this.